tests/test_http_parser.py

```diff
@@ -872,19 +872,6 @@ def test_partial_url(parser: Any) -> None:
     assert payload.is_eof()
 
 
-def test_url_parse_non_strict_mode(parser: Any) -> None:
```

Comment:
This doesn't test anything more than the below test. There is also no strict/non-strict mode as the name suggests...